CRM/Contribute/BAO/Contribution.php

```diff
@@ -4417,7 +4417,8 @@ public static function completeOrder($input, &$ids, $objects, $transaction = NUL
       $transaction = new CRM_Core_Transaction();
     }
     $contribution = $objects['contribution'];
-    $primaryContributionID = $contribution->id ?? $objects['first_contribution']->id;
+
+    $primaryContributionID = $contribution->id ?? $input['original_contribution_id'];
```

Comment:
This is only actually relevant for the membership update later on

Reply:
Not quite - it's also used earlier for the `isSingleLineItem` check. Again we can simplify further later - it's now clearer *what* that `primaryContributionID` actually represents.

Reply:
Right - neither of those uses of it actually make sense the more I think about it. So it makes more sense to work to eliminate the value

Reply:
So removing this line is still valid